kafka-cron/prometheus.yml

```diff
@@ -0,0 +1,42 @@
+## prometheus.yml ##
+
+# global settings
+global:
+  scrape_interval:     15s # Set the scrape interval to every 15 seconds. Default is every 1 minute.
+  evaluation_interval: 15s # Evaluate rules every 15 seconds. The default is every 1 minute.
+  # scrape_timeout is set to the global default (10s).
+
+alerting:
+  alertmanagers:
+    - static_configs:
+      - targets: ["alertmanager:9093"]
+
+# Load rules once and periodically evaluate them according to the global 'evaluation_interval'.
+rule_files:
+  - /etc/prometheus/alert_rules.yml
+
+# A scrape configuration containing exactly one endpoint to scrape:
+# Here it's Prometheus itself.
+scrape_configs:
+  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
+  - job_name: 'kafka'
+    metrics_path: '/metrics'
+    scrape_interval: 5s
+    static_configs:
+      - targets: ['kafka:8999']
+  - job_name: 'producers'
+    metrics_path: '/metrics'
+    scrape_interval: 5s
+    static_configs:
+      - targets: ['producer:2112']
+  - job_name: 'consumers'
+    metrics_path: '/metrics'
+    scrape_interval: 5s
+    static_configs:
+      - targets: ['cluster-a:2112', 'cluster-a-2:2112', 'cluster-b:2112', 'cluster-b-2:2112']
```

Comment:
you might like to be able to query your consumer metrics based on clusters and whether or not tasks are retries. 
you could add that info to your source metrics, or you could try using prom's relabeling (good practice to try out some relabel configs IMO). https://grafana.com/blog/2022/03/21/how-relabeling-in-prometheus-works/